staging/src/k8s.io/component-base/metrics/value.go

```diff
@@ -37,3 +39,13 @@ func (vt *ValueType) toPromValueType() prometheus.ValueType {
 func MustNewConstMetric(desc *Desc, valueType ValueType, value float64, labelValues ...string) Metric {
 	return prometheus.MustNewConstMetric(desc.toPrometheusDesc(), valueType.toPromValueType(), value, labelValues...)
 }
+
+// NewConstMetricIfNeeded is a helper of MustNewConstMetric.
+//
+// Note: If the metrics described by the desc is hidden, the metrics will not be created.
+func NewConstMetricIfNeeded(desc *Desc, valueType ValueType, value float64, labelValues ...string) Metric {
```

Comment:
Please confirm this function name, it will be exposed to downstream.

I didn't take `NewConstMetric` as the function name because Prometheus owns that. Otherwise, it may be misleading users who familiar with Prometheus.

Reply:
I like `NewLazyConstMetric` since it is more consistent with current usage.